src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadMetadata.java

```diff
@@ -120,41 +187,173 @@ public ReadMetadata read( final Kryo kryo, final Input input, final Class<ReadMe
+        public long getNBases() { return nBases; }
+        public Map<String, long[]> getReadGroupToFragmentSizeCountMap() { return readGroupToFragmentSizeCountMap; }
+
+        private void serialize( final Kryo kryo, final Output output ) {
+            final boolean refs = kryo.getReferences();
+            kryo.setReferences(false);
+            output.writeInt(readGroupToFragmentSizeCountMap.size());
+            for ( final Map.Entry<String, long[]> entry : readGroupToFragmentSizeCountMap.entrySet() ) {
+                kryo.writeObjectOrNull(output, entry.getKey(), String.class);
+                for ( final long value : entry.getValue() ) {
+                    output.writeLong(value);
+                }
+            }
+            output.writeLong(nReads);
+            output.writeLong(nBases);
+            kryo.setReferences(refs);
+        }
+
+        public static final class Serializer
+                extends com.esotericsoftware.kryo.Serializer<PartitionStatistics> {
+            @Override
+            public void write( final Kryo kryo, final Output output,
+                               final PartitionStatistics partitionStatistics ) {
+                partitionStatistics.serialize(kryo, output);
+            }
+
+            @Override
+            public PartitionStatistics read( final Kryo kryo, final Input input,
+                                             final Class<PartitionStatistics> klass ) {
+                return new PartitionStatistics(kryo, input);
+            }
+        }
+    }
+
+    /** class to track distribution of fragment lengths */
     @DefaultSerializer(ReadGroupFragmentStatistics.Serializer.class)
     public static final class ReadGroupFragmentStatistics {
-        private final float medianFragmentSize;
-        private final float medianAbsoluteDeviationFragmentSize;
+        // the distribution of fragment lengths is often quite asymmetric around the median, so we'll calculate
+        // the median deviation separately for negative and positive deviations.
+        private final int medianFragmentSize;
+        private final int medianNegativeDeviation;
+        private final int medianPositiveDeviation;
 
-        public ReadGroupFragmentStatistics( final float medianFragmentSize,
-                                            final float medianAbsoluteDeviationFragmentSize ) {
+        public ReadGroupFragmentStatistics( final long[] counts ) {
+            final long total = Arrays.stream(counts).sum();
+            long sum = 0L;
+            int medianFragmentSize = 0;
+            while ( medianFragmentSize != counts.length ) {
+                sum += counts[medianFragmentSize];
+                if ( 2*sum >= total ) break;
+                medianFragmentSize += 1;
+            }
             this.medianFragmentSize = medianFragmentSize;
-            this.medianAbsoluteDeviationFragmentSize = medianAbsoluteDeviationFragmentSize;
+
+            sum = counts[medianFragmentSize]/2;
```

Comment:
Maybe I'm being dense but I'm a little confused by this calculation. Are you assigning half of the reads at the median to the 'over/positive' group and half to the 'under/negative' group?